File: Algorithms/Bubble Sort/bubble_sort.py
#swap function
def swap(a,b):
    return b,a

#Bubble Sort Function
def bubble_sort(array):
    n = len(array) #length of the array
    #first loop
    for i in range (0,n): 
        #second loop
        for j in range(0,n-i-1): 
            #comparision
            if array[j] > array[j+1]: 
                #calling swap function
                array[j],array[j+1] = swap(array[j],array[j+1]) 
    
    return array


#Main function
#empty array creation
array = []
#getting array size from user
length = int(input("Enter the size of array : "))
#loop to append the array items into array
for i in range (length):
    array_item = int(input(f"Enter the item in array position {i+1} : "))
    array.append(array_item)
#calling the bubble sort function and storing the return value in a array
sorted_array = bubble_sort(array)
#printing the sorted list
print(f"The Sorted Array : {sorted_array}")

File: Algorithms/Bubble Sort/bubble_sort.cpp
//Bubble Sort Algorithm in c++
#include<iostream>
#include<stdlib.h>
using namespace std;

//function to print the values in the array
void display_array(int* array, int length)
{
    int i;
    for(i=0;i<length;i++)
    {
        cout<<array[i]<<" ";
    }
}

//Bubble Sort Function
int* bubble_sort(int* array, int length)
{
    int i,j,temp;
    //first loop
    for(i=0;i<length;i++)
    {
        //second loop
        for(j=0;j<length - i - 1;j++)
        {
            //comparision and swapping
            if (array[j] > array[j+1])
            {
                temp = array[j];
                array[j] = array[j+1];
                array[j+1] = temp;    
            }
        }
    }
    return array;
}
//Main function
int main()
{
    int length,element;
    //getting array size from user
    cout<<"Enter the size of array : ";
    cin>>length;
    //empty array creation
    int* array;
    array = (int*)malloc(length*sizeof(int));
    //loop to insert the array items into array
    for(int i=0;i<length;i++)
    {
        cout<<"Enter item no."<<i+1<<" in the array : ";
        cin>>element;
        array[i] = element;
    }
    //empty array creation
    int* sorted_array;
    sorted_array = (int*)malloc(length*sizeof(int));
    //calling the bubble sort function and storing the return value in a array
    sorted_array = bubble_sort(array,length);
    //printing the sorted list
    cout<<"The Sorted Array is : ";
    display_array(sorted_array, length);
    return 0;
}